### PR TITLE
feat/Add Comment Schema

### LINKED ISSUE #111: Add Comments schema

### PR DESCRIPTION
**Pull request by Navneet-Sahota**
close #111

## Types of changes

- [x] Add tags to auto close the corresponding issue.
- [x] I followed branch and commit naming convention.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] I have read the **CONTRIBUTING** document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: server/models/Comment.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  },
+  post: {
+    type: Schema.Types.ObjectId,
+    ref: 'Post',
+  },
+  user: {
+    type: Schema.Types.ObjectId,
+    ref: 'User',
+  },
+  replies: [{
```
Comment (avinash360) [inline server/models/Comment.js]: use replies as a subdocument,

ref: https://mongoosejs.com/docs/subdocs.html

File: server/models/Comment.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import mongoose from 'mongoose';
+
+const { Schema } = mongoose;
+
+const CommentSchema = new Schema({
+  comment: {
+    type: String,
+    required: [true, 'Comment message is required'],
```
Comment (avinash360) [inline server/models/Comment.js]: add a status key to comment schema

File: server/models/Comment.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    type: Schema.Types.ObjectId,
+    ref: 'Post',
+  },
+  user: {
+    type: Schema.Types.ObjectId,
+    ref: 'User',
+  },
+  replies: [{
+    reply: {
+      type: String,
```
Comment (avinash360) [inline server/models/Comment.js]: add a status key to replies schema

File: server/models/Comment.js (hunk 1 above)
Comment (Navneet-Sahota) [reply server/models/Comment.js]: Done

File: server/models/Comment.js (hunk 2 above)
Comment (Navneet-Sahota) [reply server/models/Comment.js]: Done

File: server/models/Comment.js (hunk 3 above)
Comment (Navneet-Sahota) [reply server/models/Comment.js]: Done

File: server/models/Comment.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  user: {
+    type: Schema.Types.ObjectId,
+    ref: 'User',
+  },
+  status: {
+    type: String,
+    required: true,
+    default: 'active',
+  },
+});
```
Comment (avinash360) [inline server/models/Comment.js]: add timestamps setting to replies as well.

Comment (Navneet-Sahota) [reply server/models/Comment.js]: Done

**Pull request merged**